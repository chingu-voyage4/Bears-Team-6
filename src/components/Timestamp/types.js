// @flow

import type { StartChannelAction } from '../../redux/websocketProvider/actions'
import type { SubscribeTimestampAction } from '../../redux/timestamp/actions'

export type Props = {
  startChannel: (void) => StartChannelAction,
  subscribeTimestamp: (void) => SubscribeTimestampAction,
  ts: number,
  serverStatus: string,
  channelStatus: string,
}
