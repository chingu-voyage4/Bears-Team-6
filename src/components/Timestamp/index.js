// @flow

export { connected as Timestamp } from './Timestamp'
