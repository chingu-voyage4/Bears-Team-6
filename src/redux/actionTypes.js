// @flow

import { Types as timestamp } from './timestamp/actions'
import { Types as websocketProvider } from './websocketProvider/actions'
import { Types as auth } from './auth/actions'

export const ActionTypes = {
  ...websocketProvider,
  ...timestamp,
  ...auth,
}
