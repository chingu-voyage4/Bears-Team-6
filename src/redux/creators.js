// @flow

import { Creators as timestamp } from './timestamp/actions'
import { Creators as websocketProvider } from './websocketProvider/actions'

export const Creators = {
  ...websocketProvider,
  ...timestamp,
}
