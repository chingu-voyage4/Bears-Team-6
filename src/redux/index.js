// @flow
export { Creators } from './creators'
export { ActionTypes } from './actionTypes'
export { reducers } from './reducers'
export { sagas } from './sagas'
