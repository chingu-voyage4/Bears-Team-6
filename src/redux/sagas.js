// @flow
import { all } from 'redux-saga/effects'
import type { Saga } from '../types'
import { timestamp } from './timestamp/saga'
import { websocketProvider } from './websocketProvider/saga'
import { auth } from './auth/saga'
import { userSettings } from './userSettings/saga'

const allSagas = [timestamp, websocketProvider, auth, userSettings]

export function* sagas(): Saga<void> {
  yield all(allSagas.reduce((xs, x) => xs.concat(x), []))
}
