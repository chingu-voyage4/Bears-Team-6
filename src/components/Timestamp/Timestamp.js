// @flow

import React from 'react'
import { Helmet } from 'react-helmet'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Creators } from '../../redux'
import type { Props } from './types'

export class Timestamp extends React.PureComponent<Props> {
  componentWillMount() {
    this.props.startChannel()
    this.props.subscribeTimestamp()
  }

  render() {
    const { serverStatus, channelStatus } = this.props
    return (
      <div>
        <Helmet>
          <title>Timestamp</title>
        </Helmet>
        <div>
          <p>Server status: {serverStatus} (should be off until authorized)</p>
          <p>Channel status: {channelStatus} (should be on)</p>
          <p>timestamp: {this.props.ts}</p>
        </div>
      </div>)
  }
}

const mapStateToProps = ({
  timestamp: { ts },
  websocketProvider: {
    serverStatus, channelStatus,
  },
}) => ({ ts, serverStatus, channelStatus })

const mapDispatchToProps = (dispatch) => {
  const { startChannel, subscribeTimestamp } = Creators
  return bindActionCreators(
    { startChannel, subscribeTimestamp },
    dispatch,
  )
}

export const connected = connect(mapStateToProps, mapDispatchToProps)(Timestamp)
