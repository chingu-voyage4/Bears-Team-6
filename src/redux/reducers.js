// @flow

import { combineReducers } from 'redux'
import { timestamp } from './timestamp/reducer'

export const reducers = combineReducers({
  timestamp,
})
