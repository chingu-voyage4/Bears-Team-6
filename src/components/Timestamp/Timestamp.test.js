// @flow

import React from 'react'
import { shallow } from 'enzyme'
import { createMockStore } from 'redux-test-utils'
import { connected as Timestamp } from './Timestamp'
import { reducers, ActionTypes } from '../../redux'

const initialReduxState = reducers({}, {})
const store = createMockStore(initialReduxState)

describe('Timestamp', () => {
  expect(store.isActionTypeDispatched(ActionTypes.START_CHANNEL)).toBe(false)
  expect(store.isActionTypeDispatched(ActionTypes.SUBSCRIBE_TIMESTAMP)).toBe(false)

  const wrapper = shallow(<Timestamp />, { context: { store } })
  it('renders as expected', () => {
    expect(wrapper.dive()).toMatchSnapshot()
  })

  it('dispatches START_CHANNEL action in lifecycle methods', () => {
    expect(store.isActionTypeDispatched(ActionTypes.START_CHANNEL)).toBe(true)
  })

  it('dispatches SUBSCRIBE_TIMESTAMP action in lifecycle methods', () => {
    expect(store.isActionTypeDispatched(ActionTypes.SUBSCRIBE_TIMESTAMP)).toBe(true)
  })
})
