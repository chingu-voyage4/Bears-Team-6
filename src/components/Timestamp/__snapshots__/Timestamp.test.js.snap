// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timestamp renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timestamp
    startChannel={[Function]}
    subscribeTimestamp={[Function]}
    ts={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span>
        timestamp: 
        0
      </span>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "timestamp: ",
          0,
        ],
      },
      "ref": null,
      "rendered": Array [
        "timestamp: ",
        0,
      ],
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span>
          timestamp: 
          0
        </span>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "timestamp: ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "timestamp: ",
          0,
        ],
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
